src/base/common/downloadUtil.ts

```diff
@@ -15,15 +16,46 @@ export const downloadFile = (url: string, filename?: string): void => {
   if (!url || typeof url !== 'string' || url.trim().length === 0) {
     throw new Error('Invalid URL provided for download')
   }
+
+  const inferredFilename =
+    filename || extractFilenameFromUrl(url) || DEFAULT_DOWNLOAD_FILENAME
+
+  if (isCloud) {
+    // Assets from cross-origin (e.g., GCS) cannot be downloaded this way
+    void downloadViaBlobFetch(url, inferredFilename).catch((error) => {
+      console.error('Failed to download file', error)
+    })
+    return
+  }
+
   const link = document.createElement('a')
   link.href = url
-  link.download =
-    filename || extractFilenameFromUrl(url) || DEFAULT_DOWNLOAD_FILENAME
+  link.download = inferredFilename
+  document.body.appendChild(link)
+  link.click()
+  document.body.removeChild(link)
+}
+
+/**
+ * Download a Blob by creating a temporary object URL and anchor element
+ * @param filename - The filename to suggest to the browser
+ * @param blob - The Blob to download
+ */
+export const downloadBlob = (filename: string, blob: Blob): void => {
```

Comment:
Nit: Can we use `function`?
Function expressions in the codebase are one of the causes of HMR issues and at least one memory leak.
```suggestion
export function downloadBlob(filename: string, blob: Blob): void {
```